coderd/httpapi/name_test.go

```diff
@@ -59,7 +59,64 @@ func TestValid(t *testing.T) {
 		testCase := testCase
 		t.Run(testCase.Username, func(t *testing.T) {
 			t.Parallel()
-			valid := httpapi.UsernameValid(testCase.Username)
+			valid := httpapi.NameValid(testCase.Username)
+			require.Equal(t, testCase.Valid, valid == nil)
+		})
+	}
+}
+
+func TestTemplateDisplayNameValid(t *testing.T) {
+	t.Parallel()
+	// Tests whether display names are valid.
+	testCases := []struct {
+		Username string
+		Valid    bool
+	}{
+		{"", true},
+		{"1", true},
+		{"12", true},
+		{"1 2", true},
+		{"1234 678901234567890", true},
+		{"1234567890123 5678901", true},
+		{"S", true},
+		{"a1", true},
+		{"a1K2", true},
+		{"a1b2c3 4e5f6g7h8i9j0", true},
+		{"a1b2c3d4e5f6g h8i9j0k", true},
+		{"aa", true},
+		{"aNc", true},
+		{"abcdefghijklmnopqrst", true},
+		{"abcdefghijklmnopqrstu", true},
+		{"Wow Test", true},
+
+		{" ", false},
+		{" a", false},
+		{" a ", false},
+		{" 1", false},
+		{"1 ", false},
+		{" aa", false},
+		{"aa ", false},
+		{" 12", false},
+		{"12 ", false},
+		{" a1", false},
+		{"a1 ", false},
+		{"-abcdefghijKLmnopqrstu", false},
+		{"abcdefghijklmnopqrstu-", false},
+		{"-123456789012345678901", false},
+		{"-a1b2c3d4e5f6g7h8i9j0k", false},
+		{"a1b2c3d4e5f6g7h8i9j0k-", false},
+		{"BANANAS_wow", false},
+		{"test--now", false},
```

Comment:
Why are these not valid for a human readable name? I mean, they're not pretty but a user wants what a user wants.. 😂

Reply:
Oh that's basically because of character set `_-`. I think that you're right, let's extend the regexp.

Reply:
I don't think we need to sanitize the input here, tbh. For instance, what if someone wants for the display name to be a bunch of emojis? Sanitation shouldn't really be a problem in the frontend unless someone is doing something weird with React.

Or what say you @coder/frontend? Is it bad if we don't limit names to "safe" characters?

Reply:
> For instance, what if someone wants for the display name to be a bunch of emojis? 

That is correct, but should preserve the basic sanitation level. For instance: `<space><space>` or `M<space><space><space>` doesn't make any logical sense.

> unless someone is doing something weird with React.

.. or customers are writing unsafe characters to logs (e.g. audit or access logs). Using emojis and allowing for a wide set of characters may increase the "shellshock" attack risk. This may not be applicable in our case, but still a security concern.

I guess that we perform proper sanitization, so it won't be easy to inject the XSS snippet.

Reply:
What do you think about just blocking whitespace at the beginning and ends? I think people will want to use emojis in names (we probably will too).

Reply:
It's an option to introduce this pattern, but then we need to verify if it doesn't introduce any extra risk, something like "shellshock" against audit logs. I believe that we can track this effort in a separate issue.

Reply:
Some thoughts from the FE:
 - To my knowledge, we don't sanitize other fields. It would be nice to have some consistency however we decide to move forward. Some documentation would be nice, too.
 - React [escapes](https://stackoverflow.com/questions/7381974/which-characters-need-to-be-escaped-on-html) any values embedded in JSX before rendering them, unless you use `dangerouslySetInnerHTML`. 
 - Current approach looks fine to me.

Reply:
Thank you, Kira!